File: gestao-qualidade-software/2026-2/atividades/0309/oraculo.js
// esperado: consultar("NRA1", "2026/2") devolve escolas (lista nao vazia)
// Fatia didatica. Nao e o sitio BDGETEC. Sem HTTP.

function consultar(regional, periodo) {
  if (regional === "NRA1" && periodo === "2026/1") {
    return ["Etec Sao Paulo", "Etec Zona Leste", "Etec de Artes"];
  }
  return [];
}

const PERIODO_CASO = "2026/2";
const obtido = consultar("NRA1", PERIODO_CASO);
const ok = obtido.length > 0;
console.log(ok ? "PASS" : "FAIL");
process.exit(ok ? 0 : 1);

File: gestao-qualidade-software/2026-2/atividades/0309/Oraculo.java
// esperado: consultar("NRA1", "2026/2") devolve escolas (lista nao vazia)
// Fatia didatica. Nao e o sitio BDGETEC. Sem HTTP.
// javac Oraculo.java && java Oraculo

import java.util.List;

public class Oraculo {

    static List<String> consultar(String regional, String periodo) {
        if ("NRA1".equals(regional) && "2026/1".equals(periodo)) {
            return List.of("Etec Sao Paulo", "Etec Zona Leste", "Etec de Artes");
        }
        return List.of();
    }

    public static void main(String[] args) {
        String periodoCaso = "2026/2";
        List<String> obtido = consultar("NRA1", periodoCaso);
        boolean ok = !obtido.isEmpty();
        System.out.println(ok ? "PASS" : "FAIL");
        System.exit(ok ? 0 : 1);
    }
}
